src/Common/tests/System/Diagnostics/RemoteExecutorTestBase.cs

```diff
@@ -256,6 +256,28 @@ private static RemoteInvokeHandle RemoteInvoke(MethodInfo method, string[] args,
         }
 #endif
 
+        private static MethodInfo GetMethodInfo(Delegate d)
+        {
+            // RemoteInvoke doesn't support marshaling state on classes associated with
+            // the delegate supplied (often a display class of a lambda).  If such fields
+            // are used, odd errors result, e.g. NullReferenceExceptions during the remote
+            // execution.  Try to ward off the common cases by proactively failing early
+            // if it looks like such fields are needed.
+            if (d.Target != null)
+            {
+                Type targetType = d.Target.GetType();
+                foreach (FieldInfo fi in targetType.GetFields(BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Instance | BindingFlags.DeclaredOnly))
+                {
+                    // Skip over any compiler-named fields.  Note that this logic may need to
```

Comment:
It's not really skipping over, more like asserting there aren't any such fields. Tweak comment?
Also maybe show an example of what such a name looks like?